contracts/strategies/88mph/base/EightyEightMphStrategy.sol

```diff
@@ -0,0 +1,676 @@
+     * This can be used to adjust the maturation timespan before rolling over the deposit
+     * to a new maturation date
+     */
+    function setMaturationTarget(uint64 _value) public onlyGovernance {
+        setUint256(_MATURATION_TARGET, uint256(_value));
+    }
+
+    function maturationTarget() public view returns (uint64) {
+        return uint64(getUint256(_MATURATION_TARGET));
+    }
+
+    /**
+     * @dev Resumes the ability to invest into the underlying rewards pool.
+     */
+    function continueInvesting() public onlyGovernance {
+        _setPausedInvesting(false);
+    }
+
+    /**
+     * @dev Can completely disable liquidation of rewards. This can be
+     *      useful in case of emergency exits.
+     */
+    function setSell(bool _sell) public onlyGovernance {
+        _setSell(_sell);
+    }
+
+    function setShouldRolloverDeposit(bool _value) public restricted {
+        shouldRolloverDeposit = _value;
+    }
+
+    /**
+     * @dev Set the minimum amount of reward token needed to trigger a sale (for a liquidation).
+     */
+    function setSellFloor(uint256 _sellFloor) public onlyGovernance {
+        setSellFloor(_sellFloor);
+    }
+
+    function finalizeUpgrade() external onlyGovernance {
+        // Note we don't have to transfer the vesting NFT etc. because all rewards are claimed at the upgrade
+        // fixed yield will be lost in the upgrading process however
+
+        _finalizeUpgrade();
+
+        // Reset the liquidation paths - they need to be reset manually
+        storedLiquidationPaths[mph][rewardToken()] = new address[](0);
+        storedLiquidationDexes[mph][rewardToken()] = new bytes32[](0);
+        storedLiquidationPaths[rewardToken()][underlying()] = new address[](0);
+        storedLiquidationDexes[rewardToken()][underlying()] = new bytes32[](0);
+        storedLiquidationPaths[underlying()][rewardToken()] = new address[](0);
+        storedLiquidationDexes[underlying()][rewardToken()] = new bytes32[](0);
+    }
+
+    // ---------------- Internal methods ----------------
+
+    function rewardPoolBalance() internal view returns (uint256 balance) {
+        if(depositId == 0) {
+            return 0;
+        }
+
+        uint256 maxInt = 2**256 - 1; // see https://forum.openzeppelin.com/t/using-the-maximum-integer-in-solidity/3000
+
+        // we check for the maximum amount that we could withdraw after fees
+        (balance, ) = IDInterestLens(dInterestLens).withdrawableAmountOfDeposit(IDInterest(rewardPool()), depositId, maxInt);
+
+        // the withdrawableAmountOfDeposit method used of DInterestLens calculates the withdrawable amount based on the virtualTokenTotalSupply
+        // of the deposit. the virtualTokenTotalSupply includes the interest amount that becomes available after maturation.
+        // it does however only include this interest amount AFTER maturation.
+
+        // We do not distribute the fixed yield earnings because there is no fair way to distribute them:
```

Comment:
Please take note of this. 
We earn fixed yield earnings on the strategy but I could not think of a way how this is distributed fairly to users, actually it will even end up being unfair to many. 

I thought the best thing to do here is to take it as profit sharing fee. 
For most assets the fixed yield earnings are way smaller (usually < 2%) than what we offer as value through MPH farming. 
Depositors have a lot to gain through this strategy and it was a complex one to implement, so I'd consider this fair, but please let me know if you think of anything better.